aspects/aspects_test.go

```diff
@@ -2232,3 +2234,108 @@ func (*aspectSuite) TestAspectSeveralNestedContentRules(c *C) {
 	c.Assert(err, IsNil)
 	c.Assert(val, Equals, "value")
 }
+
+func (s *aspectSuite) TestUnsetUsingMapWithNilValuesAtLeaves(c *C) {
+	databag := aspects.NewJSONDataBag()
+	aspectBundle, err := aspects.NewBundle("acc", "bundle", map[string]interface{}{
+		"foo": map[string]interface{}{
+			"rules": []interface{}{
+				map[string]interface{}{"request": "foo.a", "storage": "foo.a"},
+				map[string]interface{}{"request": "foo.b", "storage": "foo.b"},
+			},
+		},
+	}, aspects.NewJSONSchema())
+	c.Assert(err, IsNil)
+
+	asp := aspectBundle.Aspect("foo")
+	c.Assert(asp, NotNil)
+
+	err = asp.Set(databag, "foo", map[string]interface{}{
+		"a": "value",
+		"b": "other",
+	})
+	c.Assert(err, IsNil)
+
+	err = asp.Set(databag, "foo", map[string]interface{}{
+		"a": nil,
+		"b": nil,
+	})
+	c.Assert(err, IsNil)
```

Comment:
So unsetting all of the keys in a map results in an empty map remaining, but setting all of the keys in a map to `nil` results in the entire map being removed? Sorry, I'm not familiar with what options does here, so this might be what is desired.

Reply:
No, the result in both cases should be an empty map and AFAICT that's what the code does. The test is just slightly misleading because, despite the `Get("foo")`, the only rules are for "foo.a" and "foo.b" so that's what's read from the storage (which doesn't exist hence the NotFoundError). I've now added a "foo"->"foo" rule to the mocked assertion so we can actually get "foo" and see that it's an empty map and not nil. There's already an equivalent test for unset `TestAspectUnsetLeafLeavesEmptyParent`. Thank you for raising this